From 3d43b7aef3d5b7ad2ba0418d2afd657e5fedeaf5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ludovic=20Court=C3=A8s?= <ludo@gnu.org>
Date: Wed, 20 Jan 2021 11:35:10 +0100
Subject: [PATCH] import, swh: Adjust to Guile-JSON 4.5.x unspecified value
 handling.

* guix/import/cpan.scm (<cpan-release>)[home-page]: Add 'unspecified?'
case.
* guix/import/gem.scm (<gem>)[licenses]: Likewise.
(json->gem-dependency-list): Likewise.
* guix/swh.scm (<directory-entry>)[checksums]: Likewise.
---
 guix/import/cpan.scm | 3 ++-
 guix/import/gem.scm  | 5 +++--
 guix/swh.scm         | 3 ++-
 3 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/guix/import/cpan.scm b/guix/import/cpan.scm
index 514417f781..87abe9c2f1 100644
--- a/guix/import/cpan.scm
+++ b/guix/import/cpan.scm
@@ -109,6 +109,7 @@
   (home-page    cpan-release-home-page "resources"
                 (match-lambda
                   (#f #f)
+                  ((? unspecified?) #f)
                   ((lst ...) (assoc-ref lst "homepage"))))
   (dependencies cpan-release-dependencies "dependency"
                 (lambda (vector)
diff --git a/guix/import/gem.scm b/guix/import/gem.scm
index 1f6f94532e..418d716be6 100644
--- a/guix/import/gem.scm
+++ b/guix/import/gem.scm
@@ -49,6 +49,7 @@
                    ;; This is sometimes #nil (the JSON 'null' value).  Arrange
                    ;; to always return a list.
                    (cond ((not licenses) '())
+                         ((unspecified? licenses) '())
                          ((vector? licenses) (vector->list licenses))
                          (else '()))))
   (info          gem-info)
@@ -69,7 +70,7 @@
                  json->gem-dependency-list))
 
 (define (json->gem-dependency-list vector)
-  (if vector
+  (if (and vector (not (unspecified? vector)))
       (map json->gem-dependency (vector->list vector))
       '()))
 
diff --git a/guix/swh.scm b/guix/swh.scm
index 0b765cc743..f11b7ea2d5 100644
--- a/guix/swh.scm
+++ b/guix/swh.scm
@@ -348,6 +348,7 @@ FALSE-IF-404? is true, return #f upon 404 responses."
   (checksums     directory-entry-checksums "checksums"
                  (match-lambda
                    (#f  #f)
+                   ((? unspecified?) #f)
                    (lst (json->checksums lst))))
   (id            directory-entry-id "dir_id")
   (length        directory-entry-length)
-- 
2.29.2

